applications/GeoMechanicsApplication/tests/cpp_tests/test_time_stepping.cpp

```diff
@@ -0,0 +1,154 @@
+    unsigned int mSolverStrategyInitializeSolutionStepCalls = 0;
+    unsigned int mSolverStrategyPredictCalls                = 0;
+    unsigned int mSolverStrategySolveSolutionsStepCalls     = 0;
+    unsigned int mSolverStrategyFinalizeSolutionStepCalls   = 0;
+};
+
+}
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(RunReturnsNonConvergedWhenStrategyDoesNotConverge, KratosGeoMechanicsFastSuite)
+{
+    TimeStepExecutor executor;
+    auto nonconverging_strategy = std::make_shared<DummyStrategyWrapper>(false);
+    executor.SetSolverStrategy(nonconverging_strategy);
+    const auto time = 0.0;
+    KRATOS_EXPECT_EQ(TimeStepEndState::ConvergenceState::non_converged, executor.Run(time).convergence_state);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RunReturnsConvergedWhenStrategyConverged, KratosGeoMechanicsFastSuite)
+{
+    TimeStepExecutor executor;
+    auto converging_strategy = std::make_shared<DummyStrategyWrapper>(true);
+    executor.SetSolverStrategy(converging_strategy);
+    const auto time = 0.0;
+    KRATOS_EXPECT_EQ(TimeStepEndState::ConvergenceState::converged, executor.Run(time).convergence_state);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ProcessMemberFunctionsAllCalledOnce, KratosGeoMechanicsFastSuite)
+{
+    TimeStepExecutor executor;
+    auto converging_strategy = std::make_shared<DummyStrategyWrapper>(true);
+    executor.SetSolverStrategy(converging_strategy);
+    ProcessSpy spy;
+    TimeStepExecutor::ProcessRefVec process_refs{spy};
+    executor.SetProcessReferences(process_refs);
+    const auto time = 0.0;
+
+    executor.Run(time);
+
+    KRATOS_EXPECT_EQ(1,spy.NumberOfExecuteInitializeSolutionStepCalls());
+    KRATOS_EXPECT_EQ(1,spy.NumberOfExecuteFinalizeSolutionStepCalls());
+}
+
+KRATOS_TEST_CASE_IN_SUITE(SolverStrategyMemberFunctionsAllCalledOnce, KratosGeoMechanicsFastSuite)
+{
+    TimeStepExecutor executor;
+    auto converging_strategy = std::make_shared<DummyStrategyWrapper>(true);
+    executor.SetSolverStrategy(converging_strategy);
+    ProcessSpy spy;
+    TimeStepExecutor::ProcessRefVec process_refs{spy};
+    executor.SetProcessReferences(process_refs);
+    const auto time = 0.0;
+
+    executor.Run(time);
+
+    KRATOS_EXPECT_EQ(1,converging_strategy->NumberOfSolverStrategyInitializeCalls());
+    KRATOS_EXPECT_EQ(1,converging_strategy->NumberOfSolverStrategyInitializeSolutionStepCalls());
+    KRATOS_EXPECT_EQ(1,converging_strategy->NumberOfSolverStrategyPredictCalls());
+    KRATOS_EXPECT_EQ(1,converging_strategy->NumberOfSolverStrategySolveSolutionStepCalls());
+    KRATOS_EXPECT_EQ(1,converging_strategy->NumberOfSolverStrategyFinalizeSolutionStepCalls());
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ConvergingTimeStepExecutionReturnsGivenTime, KratosGeoMechanicsFastSuite)
+{
+    TimeStepExecutor executor;
+    auto converging_strategy = std::make_shared<DummyStrategyWrapper>(true);
+    executor.SetSolverStrategy(converging_strategy);
+    const auto time = 2.0;
+    executor.Run(time);
```

Comment:
I don't see why we need this call. Just calling `Run` once (at the line below) should be sufficient, right?

Reply:
indeed, done